### ISSUE TITLE
grad_norm always zero when finetuning with attn_implementation="sdpa" or attn_implementation="eager"

### ISSUE THREAD

**Issue by NebulaKoder** [QwenLM/Qwen2.5-VL#1146]
When i finetuning using the script sft.sh on V100 which do not support flash_attn, i use the sdpa or eager attention, with data_flatten = False (raise errors when set it to True). However, I found the output log with all grad_norm =  0 and all learning rate = 0, like:

{'loss': 557.5032, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.0}                                                                                                                                                                                                                        
{'loss': 70.3376, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.0}                                                                                                                                                                                                                         
{'loss': 109.2124, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.0}                                                                                                                                                                                                                        
{'loss': 144.8037, 'grad_norm': 0.0, 'learning_rate': 0.0, 'epoch': 0.0} 

I comment the line below in Qwen2.5-VL/qwen-vl-finetune/qwenvl/train/trainer.py like:
```python
#from flash_attn.flash_attn_interface import flash_attn_varlen_func
```

By the way, when i use some other GPU cards sopporting flash_attn, the values of loss, grad_norm, and learning rate ​​are all normal and expected.

I would like to what the problem is when using adpa and eager attn with the provided sft script, and thanks for any tips and help.

**Comment by NebulaKoder** [comment]
Is seems like that we used fp16 to finetune a model pretrained with bf16. And the gradient is overflow.

**Comment by yuchenlichuck** [comment]
bf16 is smart
